src/main/java/org/commcare/session/CommCareSession.java

```diff
@@ -220,12 +220,10 @@ && getCurrentFrameStepExtra(EXTRA_POST_SUCCESS) == null
             } else {
                 return null;
             }
-        } else if (entriesForCurrentCommand.size() > 1 || !entriesForCurrentCommand.elementAt(0).getCommandId().equals(currentCmd)) {
+        } else {
             //the only other thing we can need is a form command. If there's
             //still more than one applicable entry, we need to keep going
             return SessionFrame.STATE_COMMAND_ID;
-        } else {
```

Comment:
would this result in situations where we would be returning `STATE_COMMAND_ID` instead of `null` in the new code ? `null` is also symbolic to getting into a Xform so I am a bit concerned with this change.

Reply:
both of these if clause checks are handled in the other places above:
* if entries.isEmpty()
* if entries.size() == 1
  * Entry entry = getPlatform().getEntry(currentCmd);
  * if entry == null: // this handles the 2nd test `(!entries.elementAt(0).getCommandId().equals(currentCmd)`
* else  # this handles the 1st test (`entries.size() > 1`)

The only way we should return `null` (start an xform) is if there is a single entry with a command ID that matches the current command AND there isn't a post request to perform.

I think the updated logic makes this decision clearer.

Reply:
great, agree that this is a better logic flow.